drivers/SmartThings/matter-switch/src/init.lua

```diff
@@ -121,98 +270,233 @@ local function assign_child_profile(device, child_ep)
+      else -- device only supports momentary switch, no release events
+        device.log.debug("configuring for press event only")
+        supportedButtonValues_event = capabilities.button.supportedButtonValues({"pushed"}, {visibility = {displayed = false}})
+        set_field_for_endpoint(device, INITIAL_PRESS_ONLY, ep, true, {persist = true})
+      end
+
+      if supportedButtonValues_event then
+        device:emit_event_for_endpoint(ep, supportedButtonValues_event)
+      end
+      device:emit_event_for_endpoint(ep, capabilities.button.button.pushed({state_change = false}))
+    end
+  end
+end
+
+local function find_child(parent, ep_id)
+  return parent:get_child_by_parent_assigned_key(string.format("%d", ep_id))
+end
+
 local function initialize_switch(driver, device)
   local switch_eps = device:get_endpoints(clusters.OnOff.ID)
-  table.sort(switch_eps)
-  -- Since we do not support bindings at the moment, we only want to count On/Off
-  -- clusters that have been implemented as server. This can be removed when we have
-  -- support for bindings.
-  local num_server_eps = 0
-  local main_endpoint = find_default_endpoint(device)
-  for _, ep in ipairs(switch_eps) do
-    if device:supports_server_cluster(clusters.OnOff.ID, ep) then
-      num_server_eps = num_server_eps + 1
-      if ep ~= main_endpoint then -- don't create a child device that maps to the main endpoint
-        local name = string.format("%s %d", device.label, num_server_eps)
-        local child_profile = assign_child_profile(device, ep)
-        driver:try_create_device(
-          {
-            type = "EDGE_CHILD",
-            label = name,
-            profile = child_profile,
-            parent_device_id = device.id,
-            parent_assigned_child_key = string.format("%d", ep),
-            vendor_provided_label = name
-          }
-        )
+  local button_eps = device:get_endpoints(clusters.Switch.ID, {feature_bitmap=clusters.Switch.types.Feature.MOMENTARY_SWITCH})
+  local all_eps = {}
+
+  local new_profile = nil
+
+  local component_map = {}
+  local current_component_number = 2
+  local component_map_used = false
+
+  if #switch_eps > 0 or #button_eps > 0 then
+    for _,v in ipairs(switch_eps) do
+      table.insert(all_eps, v)
+    end
+    for _,v in ipairs(button_eps) do
+      table.insert(all_eps, v)
+    end
+    table.sort(all_eps)
+    -- Since we do not support bindings at the moment, we only want to count clusters
+    -- that have been implemented as server. This can be removed when we have
+    -- support for bindings.
+    local num_server_eps = 0
+    local main_endpoint = find_default_endpoint(device)
+
+    for _, ep in ipairs(all_eps) do
+      if device:supports_server_cluster(clusters.OnOff.ID, ep) or device:supports_server_cluster(clusters.Switch.ID, ep) then
+        num_server_eps = num_server_eps + 1
+        -- Configure MCD for button endpoints
+        if tbl_contains(STATIC_PROFILE_SUPPORTED, #button_eps) then
```

Comment:
I think this would break in the event that we have a more complex device type such that
```
ep1 = switch
ep2 = button
ep3 = button
```
Because then the first switch EP would be assigned to main button component, and the ep3 button would be created as a child of the first two endpoints.

Overall, I don't think we have a robust plan for handling combined switch + button device types yet. If the scope of these PR is just meant to "add support for button devices to the matter switch driver, but NOT add support for combination switch + button devices" then maybe we could simplify this by handling the button and switch eps separately and not assuming that there would be a device that supports both. 

However, if the goal is to add support for switch + button combo devices, then I think we need to reconsider how we would handle that. Right now, buttons are mostly be handled as hard-coded MCD profiles. Switch devices types (plugs, bulbs, etc.) are being handled as parent-child devices. Would a combo device have to be MCD by default? Could we mix parent child with MCD (I'm not sure if this is even possible) and have an MCD 3-button parent with a child light device (or vice versa)? Right now, I don't think this could handle devices like that properly and I think this is worth starting a discussion about - unless I have already missed this somewhere!